pyvista/core/grid.py

```diff
@@ -608,8 +608,11 @@ def __init__(
                     '    ...     origin=(10, 35, 50),\n'
                     '    ... )\n',
                 )
-        elif dimensions is not None:
-            self._from_specs(dimensions, spacing, origin)
+        else:
+            if dimensions is not None:
+                self.dimensions = dimensions
+            self.origin = origin
+            self.spacing = spacing
```

Comment:
Note that the logic here has also changed a bit. Previously, there was a bug with init where `spacing` and `origin` were only set if `dimensions` was also set. Now these values can all be initialized independently. Tests have been added for this in https://github.com/pyvista/pyvista/pull/6726/commits/7c3c3174449fe88341a22d41b6c100a3fedc5e45